Mathlib/Algebra/Order/Monoid/Unbundled/Basic.lean

```diff
@@ -314,6 +314,23 @@ end PartialOrder
 section LinearOrder
 variable [LinearOrder α] {a b c d : α}
 
+@[to_additive]
+theorem trichotomy_of_mul_eq_mul
+    [MulLeftMono α] [MulLeftStrictMono α] [MulRightMono α] [MulRightStrictMono α]
+    (h : a * b = c * d) : (a = c ∧ b = d) ∨ a < c ∨ b < d := by
+  by_cases hac : a < c
+  · right; left; exact hac
+  · by_cases hbd : b < d
+    · right; right; exact hbd
+    · left
+      simp only [not_lt] at hac hbd
+      by_contra h'
+      apply ne_of_lt _ h.symm
+      rw [not_and_or] at h'
```

Comment:
Does using `by_contra! h'` above allow to skip this line?

Reply:
unknown tactic at this stage…